### ISSUE TITLE
Create Async Pattern

### ISSUE THREAD

**Issue by kesne** [jquatier/eureka-js-client#8]
Right now, there's a problem in that we have no idea when stuff is done (outside of inspecting some state  from the outside (not good)). This is mainly a problem with a lot of the calls that require our internal caches to be filled. If you try to make those calls before there is cached data, the call throws an error. But you can never know when those are filled.

For example, the following always throws:
```javascript
var Eureka = require('eureka-js-client’);
var client = new Eureka();
var appInfo = client.getInstancesByAppId('YOURSERVICE’);
```

But this might throw, it might now. You don't really know:
```javascript
var Eureka = require('eureka-js-client’);
var client = new Eureka();
setTimeout(() => {
  var appInfo = client.getInstancesByAppId('YOURSERVICE’);
}, 1000);
```

---

Something simple we could do is add a callback to the constructor that is called when the initial registry calls are completed.

**Comment by micahr** [comment]
Seems to me that every method should be async, especially considering the http calls that are made.

**Comment by jquatier** [comment]
+1 for async all the things.

**Comment by kesne** [comment]
Currently the `getInstancesByAppId()` calls aren't async though, because they check against a local cache that is updated at an interval.

**Comment by jquatier** [comment]
True, async where we need to I guess. Currently, that's just register but I'm going to add other calls that will need it.

**Comment by kesne** [comment]
Fixed by #13.
